### PR TITLE
Commonize Typeface::familyNames

### PR DESCRIPTION
**Pull request by Schahen**
This is pretty much how I was going to approach all the places where we need basically and array of pointers of unknown size.

### CODE REVIEW

**Review by VeselovAlex — APPROVED** [review]

File: skiko/src/jvmMain/cpp/common/Typeface.cc [hunk 1]
```diff
@@ -198,24 +198,19 @@ extern "C" JNIEXPORT jboolean JNICALL Java_org_jetbrains_skia_TypefaceKt__1nGetK
...
   (JNIEnv* env, jclass jclass, jlong ptr) {
     SkTypeface* instance = reinterpret_cast<SkTypeface*>(static_cast<uintptr_t>(ptr));
     SkTypeface::LocalizedStrings* iter = instance->createFamilyNameIterator();
     std::vector<SkTypeface::LocalizedString> names;
     SkTypeface::LocalizedString name;
+    std::vector<jlong>* res = new std::vector<jlong>();
     while (iter->next(&name)) {
-        names.push_back(name);
+        res->push_back(reinterpret_cast<jlong>(new SkString(name.fString)));
+        res->push_back(reinterpret_cast<jlong>(new SkString(name.fLanguage)));
```
Comment (VeselovAlex) [inline skiko/src/jvmMain/cpp/common/Typeface.cc:210]: Will this `SkStrings` be destroyed after `DisposeArray` call?

File: skiko/src/jvmMain/cpp/common/StdVectorDecoder.cc [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#include <jni.h>
+#include "interop.hh"
+
+extern "C" JNIEXPORT jint JNICALL Java_org_jetbrains_skia_StdVectorDecoderKt_StdVectorDecoder_1nGetArraySize
+    (JNIEnv* env, jclass jclass, jlong ptr) {
+        std::vector<jlong>* vect = reinterpret_cast<std::vector<jlong> *>(ptr);
```
Comment (olonho) [inline skiko/src/jvmMain/cpp/common/StdVectorDecoder.cc:6]: Shalln't it be `std::vector<void*>` instead?

Comment (VeselovAlex) [reply skiko/src/jvmMain/cpp/common/StdVectorDecoder.cc:6]: Or even `std::vector<SkString>` to deal with string destruction

File: skiko/src/jvmMain/cpp/common/StdVectorDecoder.cc [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        return static_cast<jint>(vect->size());
+    }
+
+extern "C" JNIEXPORT void JNICALL Java_org_jetbrains_skia_StdVectorDecoderKt_StdVectorDecoder_1nDisposeArray
+    (JNIEnv* env, jclass jclass, jlong ptr) {
+        std::vector<jlong>* vect = reinterpret_cast<std::vector<jlong> *>(ptr);
```
Comment (olonho) [inline skiko/src/jvmMain/cpp/common/StdVectorDecoder.cc:12]: As @VeselovAlex mentioned, we likely shall have destructor for vector elements either explicitly I.e. stored in first vector element, or implicitly via type parameter.

File: skiko/src/nativeJsMain/cpp/StdVectorDecder.cc [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+SKIKO_EXPORT KInt org_jetbrains_skia_StdVectorDecoder__1nGetArraySize
+    (KNativePointer ptr) {
+        std::vector<KNativePointer>* vect = reinterpret_cast<std::vector<KNativePointer> *>(ptr);
+        return vect->size();
+    }
+
+SKIKO_EXPORT void org_jetbrains_skia_StdVectorDecoder__1nDisposeArray
+    (KNativePointer ptr) {
+        std::vector<KLong>* vect = reinterpret_cast<std::vector<KLong> *>(ptr);
```
Comment (olonho) [inline skiko/src/nativeJsMain/cpp/StdVectorDecder.cc:11]: Looks inconsistent.

**Review by olonho — CHANGES_REQUESTED** [review]
We must fix memory leaks.

**Review by eymar — APPROVED** [review]

File: skiko/src/nativeJsMain/cpp/StdVectorDecder.cc (hunk 4 above)
Comment (Schahen) [reply skiko/src/nativeJsMain/cpp/StdVectorDecder.cc]: fixed

File: skiko/src/jvmMain/cpp/common/interop.hh [hunk 5]
```diff
@@ -409,6 +409,7 @@ static inline jfloat fromBits(jint i) {
     return u.f;
 }
 
+<<<<<<< HEAD
```
Comment (VeselovAlex) [inline skiko/src/jvmMain/cpp/common/interop.hh:412]: merge artifact

**Pull request closed without merging**